fedimint-api/src/db/mod.rs

```diff
@@ -214,7 +301,19 @@ impl<'a> DatabaseTransaction<'a> {
     where
         K: DatabaseKey + DatabaseKeyPrefixConst,
     {
-        let key_bytes = key.to_bytes();
+        self.module_get_value(key, Vec::new()).await
+    }
+
+    pub async fn module_get_value<K>(
```

Comment:
I was hoping we don't have to introduce any new extra methods, and the `ModuleDatabaseTransaction` will just pass already concatendated key to the `DatabaseTransaction` so it doesn't even have to know about it.

In that case we can serialize the `K` straight into the `Vec<u8>` that already contains the prefix, to avoid copying everything twice (once serializing, then concatenating).